### PR TITLE
Split tactic for datatypes

### PR DESCRIPTION
**Pull request by WorldSEnder**
Implements #21 but allowing datatypes with more than one constructor: one code action per constructor is produced.

In the current state, the constructor is *not* scope checked. This works fine most of the time, but might cause troubles down the line. I'm not sure how to implement scope checking correctly: imports might be qualified, renamed or specific to some constructors only. Is there existing machinery to deal with that sort of backwards name resolution inplace of just using `occName` as I did [here](https://github.com/WorldSEnder/haskell-language-server/blob/e2b529c832732167875439633c199bb54c22ac9d/plugins/tactics/src/Ide/Plugin/Tactic.hs#L98)?

### CODE REVIEW

File: plugins/tactics/src/Ide/Plugin/Tactic.hs [hunk 1]
```diff
@@ -110,11 +114,12 @@ commandProvider HomomorphismLambdaCase =
 
 ------------------------------------------------------------------------------
 -- | A mapping from tactic commands to actual tactics for refinery.
-commandTactic :: TacticCommand -> OccName -> TacticsM ()
+commandTactic :: TacticCommand -> String -> TacticsM ()
```
Comment (isovector) [inline plugins/tactics/src/Ide/Plugin/Tactic.hs:117]: Why this change?

**Comment by isovector** [comment]
This is excellent work, thank you @WorldSEnder! 

We don't have a good story around scoping rules yet, so I wouldn't worry about it in this PR. I'm more worried about unexported data cons than unqualified occnames, as other plugins will notice the issue and suggest fixes.

I'm happy to merge this if you're happy with it!

**Review by isovector — APPROVED** [review]

File: plugins/tactics/src/Ide/Plugin/Tactic.hs (hunk 1 above)
Comment (WorldSEnder) [reply plugins/tactics/src/Ide/Plugin/Tactic.hs:117]: `OccName` is a string + a namespace. Data constructors live in a different namespace from variables, which is the reason that https://github.com/isovector/haskell-language-server/pull/30/files#diff-98860ec806a30382db306b93440066959710bb5663b2ac0fef2201e27451797fR120-R122 uses `mkDataOcc` and `mkVarOcc` respectively and https://github.com/isovector/haskell-language-server/pull/30/files#diff-98860ec806a30382db306b93440066959710bb5663b2ac0fef2201e27451797fR120-R122 can't just always use `mkVarOcc`

Comment (isovector) [reply plugins/tactics/src/Ide/Plugin/Tactic.hs:117]: Excellent catch. I'd missed that entirely, thanks!

**Comment by isovector** [comment]
I'll merge this later today!

**Comment by WorldSEnder** [comment]
And with the addition of not suggesting `I#` for goals of type `Int` by default I'm happy with the code

File: plugins/tactics/src/Ide/Plugin/Tactic.hs [hunk 2]
```diff
@@ -95,7 +96,9 @@ commandProvider Intros =
     provide Intros ""
 commandProvider Split =
   foldMapGoalType (F.fold . tyDataCons) $ \dc ->
-    provide Split $ T.pack $ occNameString $ getOccName dc
+    let cname = occNameString $ getOccName dc
+    in  requireExtensionWhen MagicHash ("#" `L.isSuffixOf` cname) $
```
Comment (isovector) [inline plugins/tactics/src/Ide/Plugin/Tactic.hs:100]: Let's use the `isAlgebraic` function to do this check instead, though I'd be open to moving this check there.

Comment (isovector) [reply plugins/tactics/src/Ide/Plugin/Tactic.hs:100]: https://github.com/isovector/haskell-language-server/blob/8772f629191816b19abd73a3fceef2210570e93a/plugins/tactics/src/Ide/Plugin/Tactic/GHC.hs#L49

Comment (isovector) [reply plugins/tactics/src/Ide/Plugin/Tactic.hs:100]: Oh sorry, I misunderstood the check here. I'd argue this behavior is so far out of the ordinary experience of most devleopers that it's not worth supporting.

Comment (WorldSEnder) [reply plugins/tactics/src/Ide/Plugin/Tactic.hs:100]: For the moment, I will use the `algebraicTyCon` check, add an ignored test case refering to #31 and tackle it in a follow up PR.

Comment (isovector) [reply plugins/tactics/src/Ide/Plugin/Tactic.hs:100]: Sounds good to me. I really appreciate your keen eye on this!
